### PR TITLE
resolve_test_tap: Fix for Circle CI

### PR DESCRIPTION
**Pull request by sjackman**
Get the tap slug from `CIRCLE_REPOSITORY_URL`.

### CODE REVIEW

File: cmd/brew-test-bot.rb [hunk 1]
```diff
@@ -132,8 +132,8 @@ def resolve_test_tap
       end
     end
 
-    return unless git_url = ENV["UPSTREAM_GIT_URL"] || ENV["GIT_URL"]
-    # Also can get tap from Jenkins GIT_URL.
+    # Get tap from Jenkins GIT_URL or Circle CIRCLE_REPOSITORY_URL.
+    return unless git_url = ENV["UPSTREAM_GIT_URL"] || ENV["GIT_URL"] || ENV["CIRCLE_REPOSITORY_URL"]
```
Comment (MikeMcQuaid) [inline cmd/brew-test-bot.rb]: I realise it was a mess before but can this go something like:
```ruby
# Get tap from Jenkins UPSTREAM_GIT_URL, GIT_URL or 
# Circle CI's CIRCLE_REPOSITORY_URL.
git_url = ENV["UPSTREAM_GIT_URL"] 
git_url ||= ENV["GIT_URL"] 
git_url ||= ENV["CIRCLE_REPOSITORY_URL"]
return unless git_url
```

Comment (sjackman) [inline cmd/brew-test-bot.rb]: I changed it to
```ruby
    # Get tap from Jenkins UPSTREAM_GIT_URL, GIT_URL or
    # Circle CI's CIRCLE_REPOSITORY_URL.
    git_url =
      ENV["UPSTREAM_GIT_URL"] ||
      ENV["GIT_URL"] ||
      ENV["CIRCLE_REPOSITORY_URL"]
    return unless git_url
```
If you prefer your suggested style above, I'll change it toute-de-suite.

**Comment by MikeMcQuaid** [comment]
Thanks @sjackman!

**Comment by sjackman** [comment]
Thanks for merging, Mike!

**Pull request merged**